mifosng-provider/src/integrationTest/java/org/mifosplatform/integrationtests/LoanDisbursementDetailsIntegrationTest.java

```diff
@@ -0,0 +1,331 @@
+				loanProductID, proposedAmount, createTranches, installmentAmount);
+		
+		System.out.println("loan with emi as 500"+this.loanTransactionHelper.getLoanDetails(this.requestSpec,this.responseSpec, loanIDWithEmi));
+		
+		System.out
+		.println("-----------------------------------LOAN CREATED WITH EMI LOANID-------------------------------------------------"
+				+ loanIDWithEmi);
+		
+		HashMap repaymentSchedule = (HashMap)this.loanTransactionHelper.getLoanDetail(this.requestSpec,this.responseSpec, loanIDWithEmi, "repaymentSchedule");
+		this.validateRepaymentScheduleWithEMI(repaymentSchedule);
+		
+		HashMap loanStatusHashMap = LoanStatusChecker.getStatusOfLoan(
+				this.requestSpec, this.responseSpec, loanIDWithEmi);
+		LoanStatusChecker.verifyLoanIsPending(loanStatusHashMap);
+
+		System.out
+				.println("-----------------------------------APPROVE LOAN-----------------------------------------------------------");
+		loanStatusHashMap = this.loanTransactionHelper
+				.approveLoanWithApproveAmount(approveDate,
+						"01 March 2014", approvalAmount, loanIDWithEmi,
+						createTranches);
+		LoanStatusChecker.verifyLoanIsApproved(loanStatusHashMap);
+		LoanStatusChecker.verifyLoanIsWaitingForDisbursal(loanStatusHashMap);
+		System.out
+				.println("-----------------------------------MULTI DISBURSAL LOAN WITH EMI APPROVED SUCCESSFULLY---------------------------------------");
+		
+
+		final Integer loanIDWithoutEmi = applyForLoanApplicationWithEmiAmount(clientID,
+				loanProductID, proposedAmount, createTranches, withoutInstallmentAmount);
+		
+		HashMap repaymentScheduleWithoutEmi = (HashMap)this.loanTransactionHelper.getLoanDetail(this.requestSpec,this.responseSpec, loanIDWithoutEmi, "repaymentSchedule");
+		System.out
+		.println("-----------------------------------LOAN CREATED WITHOUT EMI LOANID-------------------------------------------------"
+				+ loanIDWithoutEmi);
+
+		this.validateRepaymentScheduleWithoutEMI(repaymentScheduleWithoutEmi);
+
+		HashMap loanStatusMap = LoanStatusChecker.getStatusOfLoan(
+				this.requestSpec, this.responseSpec, loanIDWithoutEmi);
+		LoanStatusChecker.verifyLoanIsPending(loanStatusMap);
+
+		System.out
+				.println("-----------------------------------APPROVE LOAN-----------------------------------------------------------");
+		loanStatusHashMap = this.loanTransactionHelper
+				.approveLoanWithApproveAmount(approveDate,
+						"01 March 2014", approvalAmount, loanIDWithoutEmi,
+						createTranches);
+		LoanStatusChecker.verifyLoanIsApproved(loanStatusHashMap);
+		LoanStatusChecker.verifyLoanIsWaitingForDisbursal(loanStatusHashMap);
+		System.out
+				.println("-----------------------------------MULTI DISBURSAL LOAN WITHOUT EMI APPROVED SUCCESSFULLY---------------------------------------");
+		
+	}
+	
+	private void validateRepaymentScheduleWithEMI(HashMap repaymentSchedule){
+		assertEquals(repaymentSchedule.get("totalInterestCharged"), 160.0f);
```

Comment:
The validation does not look comprehensive.

Ideally you would want to verify each repayment for expected principal and interest split as well as the repayment date